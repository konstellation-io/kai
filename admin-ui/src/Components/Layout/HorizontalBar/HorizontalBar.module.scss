@import 'Styles/colors';

.container {
  width: 100%;
  display: flex;
  flex-direction: row-reverse;
  background-color: $bg-color-dark;
}
