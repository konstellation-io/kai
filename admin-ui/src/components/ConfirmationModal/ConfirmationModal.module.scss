@import '../../styles/colors';
@import '../../styles/mixins';

.bg {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: #000;
  opacity: 0.5;
  pointer-events: all;
  cursor: default;
  z-index: 1;
}

.container {
  padding: 14px 0 0 21px;
  background-color: $bg-color-dark;
  width: 380px;

  position: fixed;
  top: 25%;
  left: 50%;
  transform: translate(-50%, -25%);
  z-index: 10;
}

.title {
  @include font-caption;
  color: white;
}

.message {
  @include font-body;
  font-weight: 500;

  margin: 20px 21px 65px 0;
}

.comment {
  margin-bottom: 15px;
  padding-right: 21px;
}
