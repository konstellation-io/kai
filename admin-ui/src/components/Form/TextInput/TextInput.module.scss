@import '../../../styles/mixins';
@import '../../../styles/colors';

.container {
  position: relative;

  &.white,
  &.white .input {
    color: white;
  }

  &.hasClearButton .input,
  &.hasEyeButton .input {
    padding-right: 30px;
  }

  &.hasClearButton.hasEyeButton .input {
    padding-right: 60px;
  }
}

.input {
  @include form-input;

  color: $font-color-regular;

  &::-webkit-input-placeholder {
    color: $font-color-regular;
  }

  &.lockHorizontalGrowth {
    resize: vertical;
  }
}

.clearButton {
  position: absolute;
  right: 15px;
  top: 34px;
  font-size: 15px;
  line-height: 11px;
  font-weight: 500;
  cursor: pointer;
  transition: color ease 0.3s;

  &:hover {
    color: $font-color-white;
  }
}

.eyeButton {
  position: absolute;
  right: 15px;
  top: 34px;
  cursor: pointer;
  transition: color ease 0.3s, right ease 0.3s;

  &:hover {
    color: $font-color-white;
  }

  &.showClearButton {
    right: 35px;
  }
}
