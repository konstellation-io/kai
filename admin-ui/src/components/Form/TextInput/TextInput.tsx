import React, {
  useState,
  useEffect,
  MouseEvent,
  KeyboardEvent,
  FocusEvent,
  ChangeEvent,
  ReactElement
} from 'react';

import InputLabel from '../InputLabel/InputLabel';
import InputError from '../InputError/InputError';
import InputHelp from '../InputHelp/InputHelp';
import IconShow from '@material-ui/icons/RemoveRedEye';
import IconHide from '@material-ui/icons/RemoveRedEyeOutlined';
import { isFieldAnInteger } from '../check';

import cx from 'classnames';
import styles from './TextInput.module.scss';

type Props = {
  onSubmit?: Function;
  onChange?: Function;
  onBlur?: Function;
  placeholder?: string;
  label?: string;
  textArea?: boolean;
  height?: number;
  lockHorizontalGrowth?: boolean;
  limits?: {
    minWidth?: number;
    maxWidth?: number;
    minHeight?: number;
    maxHeight?: number;
  };
  error?: string;
  helpText?: string;
  showClearButton?: boolean;
  whiteColor?: boolean;
  onlyNumbers?: boolean;
  positive?: boolean;
  formValue?: string | number;
  customClassname?: string;
  hidden?: boolean;
  autoFocus?: boolean;
  maxLength?: number;
};

function TextInput({
  onSubmit = function() {},
  onChange = function() {},
  onBlur = function() {},
  placeholder = '',
  label = '',
  textArea = false,
  height = 40,
  lockHorizontalGrowth = false,
  limits = {},
  error = '',
  helpText = '',
  showClearButton = false,
  whiteColor = false,
  onlyNumbers = false,
  positive = false,
  formValue = '',
  customClassname = '',
  hidden = false,
  autoFocus = false,
  maxLength
}: Props) {
  const [value, setValue] = useState(formValue);
  const [isHidden, setIsHidden] = useState(hidden);

  useEffect(() => {
    setIsHidden(hidden);
  }, [hidden, setIsHidden]);

  useEffect(() => {
    setValue(formValue);
  }, [formValue, setValue]);

  function updateValue(newValue: string) {
    if (
      !onlyNumbers ||
      (onlyNumbers && isFieldAnInteger(newValue, positive).valid)
    ) {
      setValue(newValue);
      onChange(newValue);
    }
  }

  function onType(e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
    const newValue = e.target.value;

    updateValue(newValue);
  }

  function onKeyPress(
    e: KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>
  ) {
    if (e.which === 13 && !textArea) {
      // Enter key
      onSubmit(value);
    }
  }

  function onInputBlur(e: FocusEvent<HTMLInputElement | HTMLTextAreaElement>) {
    onBlur();
  }

  function toggleVisibility(e: MouseEvent<HTMLDivElement>) {
    setIsHidden(!isHidden);
  }

  const inputProps = {
    className: cx(styles.input, {
      [styles.error]: error !== '',
      [styles.lockHorizontalGrowth]: lockHorizontalGrowth
    }),
    value: value,
    type: isHidden ? 'password' : 'text',
    placeholder: placeholder,
    onChange: onType,
    onKeyPress: onKeyPress,
    onBlur: onInputBlur,
    style: { height },
    maxLength: maxLength
  };
  const inputElement: ReactElement =
    textArea && !isHidden ? (
      <textarea
        {...inputProps}
        data-testid="input"
        style={{ ...limits }}
        autoFocus={autoFocus}
      />
    ) : (
      <input {...inputProps} data-testid="input" autoFocus={autoFocus} />
    );
  const cleanButton =
    showClearButton && value !== '' ? (
      <div
        className={styles.clearButton}
        onClick={() => updateValue('')}
        data-testid="clear-button"
      >
        x
      </div>
    ) : (
      ''
    );
  const VisibilityIcon = isHidden ? IconShow : IconHide;
  const showEyeButton = hidden ? (
    <div
      className={cx(styles.eyeButton, {
        [styles.showClearButton]: showClearButton && value !== ''
      })}
      onClick={toggleVisibility}
    >
      <VisibilityIcon className="icon-small" />
    </div>
  ) : (
    ''
  );

  return (
    <div
      className={cx(styles.container, customClassname, {
        [styles.white]: whiteColor,
        [styles.hasClearButton]: showClearButton,
        [styles.hasEyeButton]: hidden
      })}
    >
      <InputLabel text={label} />
      {inputElement}
      {cleanButton}
      {showEyeButton}
      {error ? (
        <InputError message={error} />
      ) : (
        <InputHelp message={helpText} />
      )}
    </div>
  );
}

export default TextInput;
