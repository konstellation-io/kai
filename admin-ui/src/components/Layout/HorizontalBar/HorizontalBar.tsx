import React, { FunctionComponent } from 'react';
import styles from './HorizontalBar.module.scss';
import cx from 'classnames';

type Props = {
  children: any;
  style?: string;
};

const HorizontalBar: FunctionComponent<Props> = ({ children, style }) => (
  <div className={cx(styles.container, style)}>{children}</div>
);

export default HorizontalBar;
