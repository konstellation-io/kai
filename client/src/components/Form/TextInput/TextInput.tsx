import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import {isFieldAnInteger} from '../../Form/check'; 
import cx from 'classnames';
import styles from './TextInput.module.scss';

type Props = {
  onSubmit?: Function;
  onChange?: Function;
  placeholder?: string;
  label?: string;
  textArea?: boolean;
  height?: number;
  limits?: {
    minWidth?: number;
    maxWidth?: number;
    minHeight?: number;
    maxHeight?: number;
  };
  error?: string;
  showClearButton?: boolean;
  whiteColor?: boolean;
  onlyNumbers?: boolean;
  defaultValue?: any;
};

function TextInput({
  onSubmit = function() {},
  onChange = function() {},
  placeholder = '',
  label = '',
  textArea = false,
  height = 40,
  limits = {},
  error = '',
  showClearButton = false,
  whiteColor = false,
  onlyNumbers = false,
  defaultValue = '',
}: Props = {}) {
  const [value, setValue] = useState(defaultValue);

  useEffect(() => {
    setValue(defaultValue);
  }, [defaultValue, setValue]);

  function updateValue(newValue:any) {
    if (!onlyNumbers || (onlyNumbers && isFieldAnInteger(newValue).valid)) {
      setValue(newValue);
      onChange(newValue);
    }
  }

  function onType(e: any) {
    const value = e.target.value;

    updateValue(value);
  }

  function onKeyPress(e: any) {
    if (e.which === 13 && !textArea) {  // Enter key
      onSubmit(value);
    }
  }

  const inputProps = {
    className: cx(styles.input, {
      [styles.error]: error !== '',
    }),
    value: value,
    placeholder: placeholder,
    onChange: onType,
    onKeyPress: onKeyPress,
    style: { ...limits, height },
  };
  const inputElement = textArea ? (
    <textarea {...inputProps} data-testid="input" />
  ) : (
    <input {...inputProps} type="text" data-testid="input" />
  );
  const cleanButton = showClearButton && value !== '' ?
    <div
      className={ styles.clearButton }
      onClick={ () => updateValue('')}
      data-testid="clear-button"
    >x</div>
    : '';

  return (
    <div className={ cx(styles.container, {
      [styles.white]: whiteColor,
      [styles.hasClearButton]: showClearButton
    })}>
      <label className={styles.label} data-testid="label">
        {label.toUpperCase()}
      </label>
      {inputElement}
      {cleanButton}
      <div
        className={cx(styles.errorMessage, { [styles.show]: error !== '' })}
        data-testid="error-message"
      >
        {error}
      </div>
    </div>
  );
}

TextInput.propTypes = {
  /** Additional logic to execute after summiting */
  onSubmit: PropTypes.func,
  /** Additional logic to execute after updating the input value */
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  /** Label showed at the top of the input */
  label: PropTypes.string,
  /** If true, text input will accept multiline input */
  textArea: PropTypes.bool,
  height: PropTypes.number,
  /** limits the size textarea can grow to */
  limits: PropTypes.shape({
    minWidth: PropTypes.number,
    maxWidth: PropTypes.number,
    minHeight: PropTypes.number,
    maxHeight: PropTypes.number,
  }),
  error: PropTypes.string,
  /** Adds an 'x' button to clear the input */
  showClearButton: PropTypes.bool,
  /** font color will be brighter */
  whiteColor: PropTypes.bool,
  /** input only accept numbers */
  onlyNumbers: PropTypes.bool,
};

export default TextInput;
