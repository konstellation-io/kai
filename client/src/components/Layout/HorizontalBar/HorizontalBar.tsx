import React, { FunctionComponent } from 'react';
import styles from './HorizontalBar.module.scss';

type Props = {};

const HorizontalBar: FunctionComponent<Props> = ({ children }) => (
  <div className={styles.container}>{children}</div>
);

export default HorizontalBar;
