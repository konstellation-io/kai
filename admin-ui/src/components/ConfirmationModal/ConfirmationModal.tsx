import React, { useEffect } from 'react';
import { get } from 'lodash';
import HorizontalBar from '../Layout/HorizontalBar/HorizontalBar';
import TextInput from '../../components/Form/TextInput/TextInput';
import * as CHECK from '../../components/Form/check';
import Button from '../Button/Button';
import { useForm } from 'react-hook-form';

import styles from './ConfirmationModal.module.scss';

function verifyComment(value: string) {
  return CHECK.getValidationError([CHECK.isFieldNotEmpty(value)]);
}

type Props = {
  title: string;
  message: string;
  onAction?: Function;
  onClose?: Function;
};

function ConfirmationModal({
  title,
  message,
  onClose = function() {},
  onAction = function() {}
}: Props) {
  const { handleSubmit, setValue, register, errors } = useForm();
  useEffect(() => {
    register('comment', { validate: verifyComment });
    setValue('comment', '');
  }, [register, setValue]);

  function onSubmit(formData: any) {
    onAction(formData.comment);
  }

  function onCancelClick() {
    onClose();
  }

  return (
    <>
      <div className={styles.bg} />
      <div className={styles.container}>
        <div className={styles.title}>{title}</div>
        <div className={styles.message}>{message}</div>
        <div className={styles.comment}>
          <TextInput
            label="please, write your comment:  "
            error={get(errors.comment, 'message')}
            onChange={(value: string) => setValue('comment', value)}
            onEnterKeyPress={handleSubmit(onSubmit)}
            limits={{
              minWidth: 338,
              maxWidth: 338,
              minHeight: 105
            }}
            whiteColor
            showClearButton
            textArea
            autoFocus
          />
        </div>
        <HorizontalBar>
          <Button
            primary
            label={'YES'}
            onClick={handleSubmit(onSubmit)}
            height={30}
            style={{ width: '122px', padding: '0 0' }}
          />
          <Button
            label={'CANCEL'}
            onClick={onCancelClick}
            height={30}
            style={{ width: '122px', padding: '0 0' }}
          />
        </HorizontalBar>
      </div>
    </>
  );
}

export default ConfirmationModal;
