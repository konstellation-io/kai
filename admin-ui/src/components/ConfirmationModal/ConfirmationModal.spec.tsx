import React from 'react';
import ConfirmationModal from './ConfirmationModal';
import HorizontalBar from '../Layout/HorizontalBar/HorizontalBar';
import { shallow } from 'enzyme';
import TextInput from '../Form/TextInput/TextInput';

jest.mock('../../hooks/useInput', () => {
  // First time form will not be valid, second time it will be
  const isValid = jest.fn();
  isValid.mockReturnValueOnce(false);
  isValid.mockReturnValueOnce(true);

  return jest.fn(() => ({
    value: 'someValue',
    isValid,
    onChange() {},
    error: ''
  }));
});

const mockOnClose = jest.fn();
const mockOnAction = jest.fn();

describe('ConfirmationModal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ConfirmationModal
        title="Modal"
        message="Message"
        onClose={mockOnClose}
        onAction={mockOnAction}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('works with default props', () => {
    expect(wrapper.exists(HorizontalBar)).toBeTruthy();
    expect(wrapper.find('.title').text()).toBe('Modal');
    expect(wrapper.find('.message').text()).toBe('Message');
    expect(wrapper.exists({ label: 'YES' })).toBeTruthy();
    expect(wrapper.exists('.bg')).toBeTruthy();
  });

  it('calls onClose callback function', () => {
    wrapper.find({ label: 'CANCEL' }).simulate('click');

    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });

  it('do not call onAction callback function when form is invalid', () => {
    wrapper.find({ label: 'YES' }).simulate('click');

    expect(mockOnAction).toHaveBeenCalledTimes(0);
  });

  it('calls onAction callback function when form is valid', () => {
    wrapper
      .find(TextInput)
      .dive()
      .find('textarea')
      .simulate('change', { target: { value: 'some-message' } });

    wrapper.find({ label: 'YES' }).simulate('click');
    expect(mockOnAction).toHaveBeenCalledTimes(1);
  });
});
