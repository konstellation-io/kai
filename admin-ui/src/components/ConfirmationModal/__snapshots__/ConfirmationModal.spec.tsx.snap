// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal matches snapshot 1`] = `
<Fragment>
  <div
    className="bg"
  />
  <div
    className="container"
  >
    <div
      className="title"
    >
      Modal
    </div>
    <div
      className="message"
    >
      Message
    </div>
    <div
      className="comment"
    >
      <TextInput
        autoFocus={true}
        label="please, write your comment:  "
        limits={
          Object {
            "maxWidth": 338,
            "minHeight": 105,
            "minWidth": 338,
          }
        }
        onChange={[Function]}
        onEnterKeyPress={[Function]}
        showClearButton={true}
        textArea={true}
        whiteColor={true}
      />
    </div>
    <HorizontalBar>
      <Button
        height={30}
        label="YES"
        onClick={[Function]}
        primary={true}
        style={
          Object {
            "padding": "0 0",
            "width": "122px",
          }
        }
      />
      <Button
        height={30}
        label="CANCEL"
        onClick={[Function]}
        style={
          Object {
            "padding": "0 0",
            "width": "122px",
          }
        }
      />
    </HorizontalBar>
  </div>
</Fragment>
`;
